src/yunohost/tools.py

```diff
@@ -322,6 +322,17 @@ def tools_postinstall(operation_logger, domain, password, ignore_dyndns=False,
     else:
         dyndns = False
 
+    cron_job_file = "/etc/cron.hourly/yunohost-generate-dh-params"
+
+    command = "nice -n 19 openssl dhparam -out /etc/ssl/private/dh2048.pem -outform PEM -2 2048 -dsaparam 2> /var/log/yunohost/dhparam_generation.log && rm /etc/cron.hourly/yunohost-generate-dh-params\n"
```

Comment:
Note that a migration should be implemented so that it also gets applied to existing instances

Reply:
You also probably need to change the owner for this file like : `root:ssl-cert`

Reply:
mh ? On my instance it's root:root, and it works. If you think it's better root:ssl-cert... but it works like that (perhaps I miss something).

Reply:
That essentially depends of wether or not we will need some other service to use this DH param. For instance, SSL certificates have root:ssl-cert as permission because metronome needs access to it while not having root privileges... I don't know if dovecot or postfix or metronome or whatever service needs DH params but having ssl-cert as group for this file provides more flexibility I guess

Reply:
I say this because I'll need this for synapse :smiley:

Reply:
oh ok. Well, I was only thinking about nginx :-) For other use, ok ;-) 
Well I don't think it will be use for Dovecot or postifx...but I agree with you

Reply:
normalement corrigé.

Reply:
Note that the original comment is still valid : a migration should be implemented so that it also gets applied to existing instances

Reply:
yes, I know. I just didn't have time to do it :/ Let's go to try.

Reply:
That's okay :wink: Just be aware of it :stuck_out_tongue:

Reply:
Normally, quite done :/ Not tested as always (I'm at work for now).